mirgecom/boundary.py

```diff
@@ -35,15 +41,99 @@
 import numpy as np
 from meshmode.dof_array import thaw
 from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
-from grudge.trace_pair import TracePair
 from mirgecom.fluid import make_conserved
+from grudge.trace_pair import TracePair
+from mirgecom.inviscid import inviscid_facial_flux
 
+from abc import ABCMeta, abstractmethod
 
-class PrescribedBoundary:
-    """Boundary condition prescribes boundary soln with user-specified function.
+
+class FluidBoundary(metaclass=ABCMeta):
+    r"""Abstract interface to fluid boundary treatment.
+
+    .. automethod:: inviscid_boundary_flux
+    """
+
+    @abstractmethod
+    def inviscid_boundary_flux(self, discr, btag, cv, eos, **kwargs):
+        """Get the inviscid flux across the boundary faces."""
+
+
+class FluidBC(FluidBoundary):
+    r"""Abstract interface to boundary conditions.
+
+    .. automethod:: inviscid_boundary_flux
+    .. automethod:: boundary_pair
+    """
+
+    @abstractmethod
+    def inviscid_boundary_flux(self, discr, btag, cv, eos, **kwargs):
+        """Get the inviscid part of the physical flux across the boundary *btag*."""
+
+    @abstractmethod
+    def boundary_pair(self, discr, btag, cv, eos, **kwargs):
+        """Get the interior and exterior solution (*u*) on the boundary."""
+
+
+class PrescribedInviscidBoundary(FluidBC):
```

Comment:
I find this base class really hard to decipher (and the [viscous version](https://github.com/illinois-ceesd/mirgecom/blob/3586065551999bd8fc6da8511f9e889f363dbdcd/mirgecom/boundary.py#L136-L159) even more so) due to all of the optional callbacks. I'd like to see this get refactored at some point (maybe into a set of helper functions?). But I guess I can tolerate it going into main as is for now.